File: employee_app/src/app/models/Employee.model.ts
export class EmployeeModel {

    employeeId: number;
    name: string;
    contactNo: string;
    email: string;
    city: string;
    state: string;
    pincode: string;
    altContactNo: string;
    address: string;
    designationId: number;
    createdDate: Date | null;
    modifiedDate: Date | null;
    role: string;

    constructor() {

        this.employeeId = 0;
        this.name = '';
        this.contactNo = '';
        this.email = '';
        this.city = '';
        this.state = '';
        this.pincode = '';
        this.altContactNo = '';
        this.address = '';
        this.designationId = 0;
        this.createdDate = null;
        this.modifiedDate = null;
        this.role = '';

    }

}

File: Employee.api/Employee.api/Models/EmployeeModel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Employee.api.Models
{
    [Table("employeeTbl")]
    public class EmployeeModel
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]

        public int employeeId { get; set; }
        [Required, MaxLength(50)]
        public string name { get; set; } = string.Empty;

        [Required, MaxLength(10), MinLength(10)]
        public string contactNo { get; set; } = string.Empty;

        
        public string email { get; set; } = string.Empty;

        public string city { get; set; } = string.Empty;



        public string state { get; set; } = string.Empty;

        public string pincode { get; set; } = string.Empty;


        public string? altContactNo  { get; set; } 


        public string address { get; set; } = string.Empty;


        public int designationId { get; set; }

        public DateTime? createdDate { get; set; }

        public DateTime? modifiedDate { get; set; }

        public string? role  { get; set; }
        
        
    }



    public class LoginModel
    {
        [Required]
        [EmailAddress]
        public string email { get; set; } = string.Empty;

        [Required]
        public string contactNo { get; set; } = string.Empty;
    }




}
